ledger/error/src/main/scala/com/daml/error/generator/ErrorCodeDocumentationGenerator.scala

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.error.generator
+
+import java.lang.reflect.Modifier
+import com.daml.error.ErrorCode
+import com.daml.error.generator.ErrorCodeDocumentationGenerator.{
+  explanationTypeName,
+  resolutionTypeName,
+}
+import com.daml.error.{Explanation, Resolution}
+import org.reflections.Reflections
+
+import scala.reflect.runtime.{universe => ru}
+import scala.jdk.CollectionConverters._
+
+/** Utility that indexes all error code implementations.
+  *
+  * @param prefix The classpath prefix that should be scanned for finding subtypes of [[ErrorCode]].
+  */
+case class ErrorCodeDocumentationGenerator(prefix: String = "com.daml") {
+
+  def getDocItems: Seq[DocItem] = {
+    val errorCodes = getErrorCodeInstances
+
+    errorCodes.view.map(_.id).groupBy(identity).collect {
+      case (code, occurrences) if occurrences.size > 1 =>
+        sys.error(
+          s"Error code $code is used ${occurrences.size} times but we require each error code to be unique! " +
+            s"Make these error code unique to make this assertion run through"
+        )
+    }
+
+    errorCodes.map(convertToDocItem).sortBy(_.code)
+  }
+
+  private def getErrorCodeInstances =
```

Comment:
Do we care of adding empty parameter list as this theoretically return different results on each call (because a new class has been loaded in the meantime?)

Reply:
I think that `Reflections` scans the entire classpath so the results are (hopefully) deterministic.